File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        min_price = 10000000
        max_profit = 0

        for price in prices:
            if price < min_price:
               min_price = price
            else:
                price = price - min_price
                if price > max_profit:
                    max_profit = price
        return max_profit               
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.dart
class Solution {
  int maxProfit(List<int> prices) {
    int buy_price = prices[0];
    int profit = 0;
    for(int i = 1;  i < prices.length ; i++) {
        if(prices[i] < buy_price) buy_price = prices[i];
         profit = max(profit, prices[i] - buy_price);    
    }
   return profit;
  }
} 